### ISSUE TITLE
Welcome

### ISSUE THREAD

**Issue by jbenet** [ipfs/blockchain-data#1]
Carrying over from https://github.com/ipfs/faq/issues/14 (figured a repo with multiple threads made more sense)

**Comment by jbenet** [comment]
@shea256 @muneeb-ali @compleatang @prestonjbyrne @buchmanster @taoeffect

**Comment by jbenet** [comment]
what's Devon Read's github?

**Comment by prestonjbyrne** [comment]
@ebuchman

**Comment by jbenet** [comment]
@guylepage3

**Comment by jbenet** [comment]
@john-light

**Comment by prestonjbyrne** [comment]
@androlo @dennismckinnon

**Comment by jbenet** [comment]
Everyone, it sounds like all our work is converging faster than expected. Let's discuss here-- it's open, public, and has much more space than twitter.

**Comment by jbenet** [comment]
@muneeb-ali said

> Some background on the data-storage of Blockstore:
> 
> https://github.com/openname/blockstore/wiki/Data-Storage

**Comment by muneeb-ali** [comment]
Yes, so I'll be adding more info/diagrams on the data-storage page. @jcnelson has started looking at a whitepaper today that will also discuss this (and we can try to keep the Github page updated with the whitepaper)

**Comment by shea256** [comment]
hola

**Comment by jbenet** [comment]
watch this Repo

**Comment by DevonJames** [comment]
'ello ;)

**Comment by whyrusleeping** [comment]
checking in for updates

**Comment by shea256** [comment]
Here's the proposal for blockstore to support the packing of many files into a single Bitcoin transaction: https://github.com/openname/blockstore/issues/81

Would love all your thoughts on this. @DevonJames this is especially relevant to your concerns about high fees.

**Comment by jbenet** [comment]
@muneeb-ali @shae256 it would be useful to setup IPFS as a blockstore mirror sometime soon. You can use it as a key-value store.

**Comment by shea256** [comment]
Let's do it. Also watch out for that typo!

**Comment by muneeb-ali** [comment]
Adding a little historic context to this conversation. The DHT community in the early 2000s got really excited about things like [replacing DNS](http://nms.lcs.mit.edu/papers/sfr-nsdi04.pdf), [replacing lookups on the Internet](http://www.cs.berkeley.edu/~istoica/papers/i3-sigcomm02.pdf), decentralized storage, and so on. But none of that really materialized.

This is the "second wave" in a way, to take a stab at these problems and we should be aware of what is different now compared to earlier. One thing is obviously the blockchain ...

**Comment by DevonJames** [comment]
we're all just updating ourselves on IPFS over here, and really quite stoked

**Comment by DevonJames** [comment]
changes our thinking about a lot of stuff - primarily how much info should go in the chain since IPFS looks to be way faster than the mainline DHT

**Comment by DevonJames** [comment]
@muneeb-ali ,@shea256 - since your time with onename and now working on blockstore, whats your take so far on the specialization of namecoin vs security (even if it requires somewhat tweaked mechanisms) of bitcoin?

**Comment by DevonJames** [comment]
oh, and @jbenet can you explain how filecoin and IPFS relate to one another?

**Comment by DevonJames** [comment]
we're really weighing the question of a specialized alt coin vs changing data structure to use BTC blockchain, but IPFS as a replacement for mainline DHT makes it more compelling (much faster discovery and loading) but then how does filecoin play into the mix?

**Comment by DevonJames** [comment]
plus, of course, this: https://github.com/openname/blockstore/issues/81 to reduce the cost of submissions

**Comment by ebuchman** [comment]
couple things:
1) would love to see a standard/initiative emerge to get a single DHT backbone for bootstrapping all p2p protocols. We could eg. use DNS to map a url to some "protocol id", and then find peers for that protocol on the DHT. You know, put an end to the bootstrap problem once and for all. We might need to pester OS/browser vendors to get onboard so we can easily bootstrap onto the DHT itself. A "one bootstrap, one time, to rule them all" approach

2) I'm very interested in building a minimal blockchain in javascript entirely on top of ipfs. That is, put a directory of js files and a genesis block on ipfs. A user pulls the js, its presumably a node app, say, it could even be python we could do it 3 languages as a proof of concept. then all blocks/txs could just be ipfs blobs. the trick will be how to handle the non-persistent network messaging, like for getBlocks or getTxs. @jbenet does IPFS support any form of custom overlay networks so I can write a p2p protocol above it?

I'm still in SF. Maybe we can plan a few days to get together a few of us and try and hack on something like this?

**Comment by whyrusleeping** [comment]
@ebuchman you can build an overlay network on ipfs very easily. I have some examples here: http://gateway.ipfs.io/ipfs/QmTkzDwWqPbnAh5YiV5VwcTLnGdwSNsNTn2aDxdXBFca7D/example#/ipfs/QmX6gcmX2vy2gs5dWB45w8aUNynEiqGhLayXySGb7RF2TM/readme.md

**Comment by DevonJames** [comment]
@jbenet @prestonjbyrne thoughts on webtorrent? turning users on websites into seeders could really be a big deal. https://instant.io/

**Comment by whyrusleeping** [comment]
@DevonJames we plan on having an implementation of ipfs in javascript so we can do exactly that!

**Comment by jbenet** [comment]
@ebuchman 

>  would love to see a standard/initiative emerge to get a single DHT backbone for bootstrapping all p2p protocols.

entirely agreed. it's not trivial because DHTs can be hard to get right for each different use case. for example, we didn't use MainlineDHT because it wouldn't be nice to torrent clients. we dump _lots_ of records into our DHT.

> @jbenet does IPFS support any form of custom overlay networks so I can write a p2p protocol above it?

as @whyrusleeping said, yep. can mount your own protocols on top of ipfs. can use it as a `libp2p` of sorts. (we actually plan to extract this out in time).

> I'm still in SF. Maybe we can plan a few days to get together a few of us and try and hack on something like this?

Happy to. i'll be in SF wed and thu for 
- fluentconf.com/javascript-html-2015/public/schedule/detail/43686
- www.meetup.com/EthereumSiliconValley/events/221350594/
  maybe after either. Or come down to Palo Alto friday or next week. I'm afk this weekend though.

**Comment by jbenet** [comment]
@DevonJames 

> can you explain how filecoin and IPFS relate to one another?

Filecoin is a sister protocol of IPFS. you can use either without the other. We're focused on getting IPFS solid first.

> IPFS as a replacement for mainline DHT makes it more compelling (much faster discovery and loading) 

Yep that's why we made it.

**Comment by muneeb-ali** [comment]
@DevonJames wrote:

> @muneeb-ali ,@shea256 - since your time with onename and now working on blockstore, whats your take so far on the specialization of namecoin vs security (even if it requires somewhat tweaked mechanisms) of bitcoin?

I've added more notes on the choice of blockchain here:
https://github.com/openname/blockstore/wiki/Blockchain 

Let's try to define how exactly are we comparing the different blockchain options (what metrics do we care about) and then it'll be easier to compare Namecoin vs Bitcoin vs something else. 

As you can probably guess, out of the metrics I've listed. The Bitcoin blockchain clearly outperforms Namecoin on every single metric.

**Comment by ebuchman** [comment]
@jbenet 

> it's not trivial because DHTs can be hard to get right for each different use case. 

Sure, and that's fine, but I'm talking about a DHT for a single common use case of finding peers for every other DHT. So eg someone wants to connect to IPFS. They know the ipfs protocol url. Say its "protocol.ipfs.io". DNS query returns a hash. Then they look up that hash on this global dht to find IPFS peers. then they can connect to those peers and join IPFS' dht. and so on.  Would be really nice if some international standard was set for bootstrapping seeds in p2p protocols is all.

> can mount your own protocols on top of ipfs.

Awesome.

I'll try and swing by tmrw/thurs.

**Comment by jbenet** [comment]
@ebuchman 

> Sure, and that's fine, but I'm talking about a DHT for a single common use case of finding peers for every other DHT. So eg someone wants to connect to IPFS. They know the ipfs protocol url. Say its "protocol.ipfs.io". DNS query returns a hash. Then they look up that hash on this global dht to find IPFS peers. then they can connect to those peers and join IPFS' dht. and so on. Would be really nice if some international standard was set for bootstrapping seeds in p2p protocols is all.

very much agreed :+1:

**Comment by denisnazarov** [comment]
Wanted to add https://github.com/mine-code/canonical-content-registry to the conversation.

CCR goal is to write a lot of metadata to a DHT timestamped with the blockchain, with media retrieval as a necessity down the line.

@moudy @jessewalden
